ARTstor.js

```diff
@@ -1,131 +1,899 @@
+    
+    Zotero is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+    
+    You should have received a copy of the GNU Affero General Public License
+    along with Zotero.  If not, see <http://www.gnu.org/licenses/>.
+    
+    ***** END LICENSE BLOCK *****
 */
 
-function getData(doc, checkOnly) {
-	var data = {},
-		widgets = doc.getElementsByClassName('MetaDataWidget'),
-		topWidget = null, topZIndex;
-	
-	// Dtermine the top metadata window
-	for (var i=0; i<widgets.length; i++) {
-		
-		var zindex = doc.defaultView.getComputedStyle(widgets[i], null)
-			.getPropertyValue('z-index');
-		if (!zindex && zindex !== 0) continue;
-		
-		if (!topWidget || topZIndex < zindex) {
-			topWidget = widgets[i];
-			topZIndex = zindex;
-		}
-	}
-	
-	if (!topWidget) return false;
-	
-	var fields = topWidget.getElementsByClassName('field'),
-		found = false;
-	for (var i=0; i<fields.length; i++) {
-		var field = ZU.trimInternal(fields[i].textContent);
-		
-		var value = fields[i].nextElementSibling;
-		if (!value || !value.classList.contains('data')) continue;
-		value = ZU.unescapeHTML(ZU.cleanTags(value.innerHTML));
-		
-		if (!field || !value.trim()) continue;
-		
-		if (checkOnly) return true;
-		
-		found = true;
-		data[field] = value;
-	}
-	
-	return found ? data : false;
+/**
+    detectWeb is run to determine whether item metadata can indeed be retrieved from the webpage. 
+    The return value of this function should be the detected item type (e.g. “journalArticle”, 
+    see the overview of Zotero item types), or, if multiple items are found, “multiple”. 
+**/
+function detectWeb(doc, url) {
+    if (url.match(/\/iv2/)) {
+        // Image viewer window
+        return "artwork";
+    } else if (url.match(/\#3\|/)) {
+        // Thumbnail window page
+        if ((doc.getElementsByClassName('MetaDataWidgetRoot') != null) && (doc.getElementsByClassName('MetaDataWidgetRoot').length > 0)) {
+            // There are multiple metadata windows visible
+            return "artwork";
+        } else if ((doc.getElementById("floatingPlaceHolder") != null) && (doc.getElementById("floatingPlaceHolder").style.display == "block")) {
+            // Don't capture date if small window is present
+            return false;
+        } else if ((doc.getElementById("thumbNavSave1") != null) && (doc.getElementById("thumbNavSave1").style.display == "block")) {
+            // Don't capture data if image group window is in editing state.
+            return false;
+        } else if ((doc.getElementById("ssContentWrap") != null) && (doc.getElementById("ssContentWrap").style.display == "inline")) {
+            // Don't capture data if slide show window is present
+            return false;
+        } else {
+            if (url.match(/zMode/)) {
```

Comment:
Actually, you [originally](https://github.com/venturi2k/translators/commit/cbe856c399209756434408d8fc496e95d4be615c#diff-2bbd96777b59793a19b2dfc8752323ecR58) had it as
```
} else
if (....) {
```
which JSBeautifier (correctly) identified as
```
} else {
  if (...) {
```
Technically they accomplish the same thing, but we'd like to stick to more typical formatting and not have to worry about some unanticipated behavior.